### ISSUE TITLE
Can't change lime-packages branch when old branch has been deleted

### ISSUE THREAD

**Issue by ilario** [libre-mesh/lime-build#4]
Yesterday I compiled from lime-packages branch sandbox/vlanzero.
Then today I tried to compile from branch develop but I got an error.
The problem is that the sandbox/vlanzero has been deleted (branches are deleted quite often from lime-packages) but lime-build tries to fetch the old branch even if a new one is specified.

`cd lime-build`
`git checkout develop`
`git pull`
`make T=tl-wdr3600 LIME_GIT_BRANCH=develop UPDATE=1 build`

`Updating LiMe repository
(cd build/lime-packages && git pull && git checkout develop && git pull origin develop)
Your configuration specifies to merge with the ref 'refs/heads/sandbox/vlanzero'
from the remote, but no such ref was fetched.
Makefile:194: recipe for target 'checkout' failed
make: *** [checkout] Error 1`

Seems that this is caused by the first `git pull` on this [line](https://github.com/libre-mesh/lime-build/commit/9617394501ef186a90d584e0c8491df34115cc7a#diff-b67911656ef5d18c4ae36cb6741b7965R180).

**Comment by G10h4ck** [comment]
as a temporary workaround you can run
make clean

**Comment by ilario** [comment]
I already solved removing the `git pull` in the pointed line. 
If you want I can make a pull request with this.

**Comment by G10h4ck** [comment]
Seems a nice idea to have a pull request fixing this, although I am not lime-build expert Pau and Gui should review your PR

Cheers!

**Comment by ilario** [comment]
Today I had somewhat the opposite problem, I moved from develop to testing/16.07 lime-build branch without running a make clean.
So:

```
cd lime-build
git pull
git checkout testing/16.07
make T=ar71xx P=generic J=3 UPDATE=1
```

and the output was:

```
Using LiMe Git repository git://github.com/libre-mesh/lime-packages.git
cd build/lime-packages && git checkout testing/16.07 && git pull
error: pathspec 'testing/16.07' did not match any file(s) known to git.
Makefile:166: recipe for target 'checkout' failed
make: *** [checkout] Error 1
```

This happened because a git checkout to the new branch was issued before the pull, so the branch didn't exist yet.
A solution (that doesn't trigger the old problem object of this issue) would be to replace the two occurrences of 
`git checkout $(LIME_GIT_BRANCH)`
in Makefile with
`git checkout develop && git pull && git checkout $(LIME_GIT_BRANCH)`
as the develop branch will always exist (I believe).

**Comment by p4u** [comment]
|I think

cd build/lime-packages && git pull origin ||testing/16.07| && |git
checkout testing/16.07|

Should solve the issue, but I have not tested it.
If you are able to test it and works, please send the PR to lime-build
develop branch.

Thanks.

**Comment by ilario** [comment]
Seems ok, but I suspect we will find more corner cases breaking this in the future.

**Comment by altergui** [comment]
On 25/07/16 16:06, Ilario Gelmetti wrote:

> Seems ok, but I suspect we will find more corner cases breaking this in
> the future.

git remote update && git checkout whatever-new-branch

should solve it in a "proper" way

**Comment by ilario** [comment]
Should have been fixed in fb64a79.
